wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -0,0 +1,46 @@
+{% load wagtailadmin_tags i18n %}
+{% with id=title|cautious_slugify %}
+    <template data-dialog>
+        <div
+            id="{{ id }}"
+            aria-labelledby="title-{{ id }}"
+            aria-hidden="true"
+            class="{{ class_name }} w-fixed w-inset-0 w-z-dialog w-flex aria-hidden:w-hidden w-p-4"
+        >
+            {# Overlay #}
+            <div data-a11y-dialog-hide class="w-fixed w-inset-0 w-bg-black/40"></div>
+
+            {# Dialog #}
+            <div class="w-w-full w-max-w-2xl sm:w-w-[600px] w-m-auto w-z-dialog w-relative w-bg-white w-shadow-dialog w-rounded-md w-overflow-hidden w-animate-fade-in">
+
+                {# Close button #}
```

Comment:
```suggestion
                {% block closebutton %}
```

If we used blocks for these, including subtitle message and others we solve two two problems. 1 - we can define the section instead of comments and 2 - when using template inheritance these blocks can be overridden or extended. 

Note that template overrides are a documented way to do some kinds of customisations and it seems reasonable to provide these here.

Reply:
That makes sense and I'm totally happy to add those in for sure, but as far as I know this component was intended to be used as a template tag itself (not as an include or extended from). 
I'm also concerned if someone where to extend this template, it would be broken because of the unclosed `div && template` tags

Reply:
fair enough, I think for the scope of this it makes sense to leave that use case out for now.